crates/test-utils/src/bin/dump-katana.rs

```diff
@@ -35,11 +36,11 @@ async fn main() {
             .dump_state()
             .expect("Failed to call dump_state on Katana state");
 
-        let state = serde_json::to_string(&dump_state).expect("Failed to serialize state");
-
         // Dump the state
         std::fs::create_dir_all(".katana/").expect("Failed to create Kakata dump dir");
-        std::fs::write(".katana/dump.json", state).expect("Failed to write dump to .katana/dump.json");
+        let katana_dump_file =
+            File::options().create_new(true).read(true).write(true).append(true).open(".katana/dump.json").unwrap();
+        serde_json::to_writer(katana_dump_file, &dump_state);
```

Comment:
if there's no impact on speed, we can maybe use `to_writer_pretty` here as well

Reply:
Hmm, don't this will be any big performance prob, so we can do that!

Reply:
Done [here](https://github.com/kkrt-labs/kakarot-rpc/pull/562/commits/f0a1c75e44c32990f28ea0f99131de8a69b641f0).